src/lib/php/libschema.php

```diff
@@ -515,12 +515,9 @@ function dropViews($catalog)
     while ($row = $this->dbman->fetchArray($result)) {
       $View = $row['view_name'];
       $table = $row['table_name'];
-      if (empty($this->schema['TABLE'][$table])) {
-        continue;
-      }
       $column = $row['column_name'];
-      if (empty($this->schema['TABLE'][$table][$column])) {
-        $sql = "DROP VIEW \"$View\";";
+      if (empty($this->schema['TABLE'][$table]) || empty($this->schema['TABLE'][$table][$column])) {
+        $sql = "DROP VIEW IF EXISTS\"$View\";";
```

Comment:
Instead of adding `IF EXISTS` (which we know exists since we just queried it with select above), why not filter out the views created by the system by joining `information_schema.views` instead? Something like following:

```sql
SELECT view_name,v.table_name,column_name
FROM information_schema.view_column_usage AS vcs
INNER JOIN information_schema.views AS v
  ON vcs.view_name = v.table_name
WHERE vcs.table_catalog='$catalog'
  AND v.table_schema = 'public'
ORDER BY view_name,vcs.table_name,column_name;
```